libpdb/client.go

```diff
@@ -50,40 +59,75 @@ func (c *Client) Ping() bool {
 	}
 }
 
-func (c *Client) CreateTopic() (*Topic, error) {
-	password := ""
-	handle, err := NewTopic(password)
-	//@todo
-	return handle, err
-}
+func (c *Client) Publish(handle *Topic, data []byte) error {
+	config := c.config.Load().(ClientConfig)
+	write_args, err := handle.GeneratePublish(config.CommonConfig, data)
+	if err != nil {
+		return err
+	}
 
-func (c *Client) Publish(data []byte) bool {
-	//@todo using EnqueueWrite
-	return true
+	c.msgReqMan.EnqueueWrite(write_args)
+	return nil
 }
 
-func (c *Client) Subscribe(handle *Topic) bool {
-	//@todo using EnqueueRead
-	return true
+func (c *Client) Poll(handle *Subscription) chan []byte {
+	// Check if already subscribed.
+	c.subscriptionMutex.Lock()
+	for x := range c.subscriptions {
+		if &c.subscriptions[x] == handle {
+			c.subscriptionMutex.Unlock()
+			return nil
+		}
+	}
+	c.subscriptions = append(c.subscriptions, *handle)
+	c.subscriptionMutex.Unlock()
+
+	return handle.Updates
 }
 
-func (c *Client) PublishTrace() uint64 {
-	config := c.config.Load().(ClientConfig)
-	req := &common.WriteArgs{}
-	req.ReplyChan = make(chan *common.WriteReply)
-	c.msgReqMan.generateRandomWrite(config, req)
-	c.msgReqMan.EnqueueWrite(req)
-	reply := <-req.ReplyChan
-	return reply.GlobalSeqNo
+func (c *Client) Done(handle *Subscription) bool {
```

Comment:
how, life-cycle-wise, would one stop receiving incoming messages from the subscription? The go abstraction for channels, as i understand it, means that the producer doesn't know when a channel has been close'ed, and will block when trying to write in such a case.